tests/ecosystem/upgrade/test_noobaa.py

```diff
@@ -159,3 +169,48 @@ def test_buckets_after_upgrade(upgrade_buckets, mcg_obj_session):
     """
     for bucket in mcg_obj_session.read_system().get('buckets'):
         assert bucket.get('mode') == BS_OPTIMAL
+
+
+@pre_upgrade
+def test_start_upgrade_mcg_io(mcg_workload_job):
+    """
+    Confirm that there is MCG workload job running before upgrade.
+    """
+    job_status = None
+    job_name = mcg_workload_job.ocp.resource_name
+    # wait a few seconds for fio job to start
+    logger.info(f"Checking number of active pods for job {job_name}")
+    for i in range(0, 5):
+        job = mcg_workload_job.ocp.get(
+            resource_name=job_name,
+            out_yaml_format=True
+        )
+        # 'active' attribute holds information about how many pods are running
+        job_status = job['items'][0]['status']['active']
```

Comment:
Have you considered replacing the direct access with several `.get()` calls?
Ditto for line 212